openlibrary/core/lending.py

```diff
@@ -495,13 +495,53 @@ def _get_ia_loan(identifier, userid):
 
 def get_loans_of_user(user_key):
     """TODO: Remove inclusion of local data; should only come from IA"""
+    if 'env' not in web.ctx:
+        """For the get_cached_user_loans to call the API if no cache is present,
+        we have to fakeload the web.ctx
+        """
+        delegate.fakeload()
+
     account = OpenLibraryAccount.get(username=user_key.split('/')[-1])
 
     loandata = web.ctx.site.store.values(type='/type/loan', name='user', value=user_key)
     loans = [Loan(d) for d in loandata] + (_get_ia_loans_of_user(account.itemname))
+    get_cached_loans_of_user.memcache_set(
```

Comment:
```suggestion
    """The reason we unset the entry is, in most cases when we want to fetch ground truth loans data it's because we're making an action on the book like return or borrow which affects one's loans. Therefore, we don't want to cache the response (which is pre-that-action) that comes back, instead, we unset so the action completes and the next time we visit a page that needs loans, we fetch loans from ground truth and cache that.
    """
    get_cached_loans_of_user.memcache_set(
```